lib/spack/spack/patch.py

```diff
@@ -119,7 +119,8 @@ def apply(self, stage):
             patch_stage.fetch()
             patch_stage.check()
             patch_stage.cache_local()
-            patch_stage.expand_archive()
+            if self.url.lower().endswith('.tar'):
+                patch_stage.expand_archive()
```

Comment:
If they are uncompressed, what is the value in using a `.tar` at all? Are they multiple files?